mlflow/utils/search_utils.py

```diff
@@ -241,22 +241,21 @@ def does_run_match_clause(cls, run, sed):
                                       "not one of '%s" % (comparator,
                                                           cls.VALID_METRIC_COMPARATORS),
                                       error_code=INVALID_PARAMETER_VALUE)
-            metric = next((m for m in run.data.metrics if m.key == key), None)
+            metric = next((m for m in run.data.metrics.values() if m.key == key), None)
```

Comment:
why not:
``` 
run.data.metrics[m.key]  if m.key in run.data.metrics else None
```
?

Reply:
Your suggestion seems clearer to me, will make that change :)